mislabeled/probe/_influence.py

```diff
@@ -135,47 +71,31 @@ def __call__(self, estimator, X, y):
             n x 1 array of the per-examples gradients
         """
 
-        grad_log_loss = estimator.predict_proba(X)
-        grad_log_loss[np.arange(len(y)), y] -= 1
+        grad_log_loss = estimator.grad_y(X, y)
 
         return norm2(grad_log_loss) * norm2(X)
 
 
-class L2GradNorm2(Minimize):
-    """The squared norm of individual gradients wrt parameters in a linear
-    model. This is e.g. used (in the case of deep learning) in the TracIn paper [1]_.
-
-    NB: it assumes that the loss used is the l2 loss a.k.a. the mean squared error
-
-    References
-    ----------
-    ..[1] Pruthi, G., Liu, F., Kale, S., & Sundararajan, M.\
-        "Estimating training data influence by tracing gradient descent." NeurIPS 2020
-    """
-
-    @linear
-    def __call__(self, estimator, X, y):
-        grad_l2_loss = estimator.predict(X) - y
-        if grad_l2_loss.ndim == 1 or grad_l2_loss.shape[1] == 1:
-            grad_l2_loss = grad_l2_loss.reshape(-1, 1)
-        return norm2(grad_l2_loss) * norm2(X)
-
-
 class Representer(Minimize):
     """Representer values"""
 
     @linear
     def __call__(self, estimator, X, y):
-        grad_log_loss = estimator.predict_proba(X)
-        grad_log_loss_observed = grad_log_loss[np.arange(len(y)), y] - 1
+        grad = estimator.grad_y(X, y)
+        # grad observed
+        if estimator._is_binary():
+            grad_observed = grad[:, 0]
+        else:
+            grad_observed = grad[np.arange(X.shape[0]), y]
 
-        return np.abs(grad_log_loss_observed) * norm2(X)
+        # print(grad.shape,norm2(X).shape )
+        return np.abs(grad_observed) * norm2(X)
 
 
 class L2Representer(Minimize):
     """L2 Representer values"""
 
     @linear
     def __call__(self, estimator, X, y):
-        grad_l2_loss = estimator.predict(X) - y
-        return np.abs(grad_l2_loss) * norm2(X)
+        grad = estimator.grad_y(X, y)
+        return np.abs(grad).sum(axis=1) * norm2(X)
```

Comment:
would be nice if L2Representer could be merged with Representer as with other probes, but it's just a nitpick.